elyra/pipeline/component.py

```diff
@@ -381,6 +381,10 @@ def determine_type_information(self, parsed_type: str) -> SimpleNamespace:
                 elif data_type in ['list', 'set', 'array', 'arr']:
                     data_type = "list"
                     default_value = []
+                else:
+                    # TODO: Unknown container type, surface error?
+                    data_type = "unknown"
+                    default_value = None
```

Comment:
I think this `else:` block needs to be removed.  The non-container types are handled in the `else` block associated with exhausting the `for` lopp (which is still a hard thing for me to remember with python).  `option` is one of the supported container types by virtue of making into the first `if` statement.

Instead, the `elif` should be an `else` (if that's what the CodeQL complaint was about).
```suggestion
                else:  # data_type is one of ['list', 'set', 'array', 'arr']
                    data_type = "list"
                    default_value = []
```